src/utils.rs

```diff
@@ -84,3 +84,11 @@ pub fn span_help_and_lint(cx: &Context, lint: &'static Lint, span: Span,
         cx.sess().span_help(span, help);
     }
 }
+
+/// return the base type for references and raw pointers
+pub fn walk_ty<'t>(ty: ty::Ty<'t>) -> ty::Ty<'t> {
```

Comment:
Can we rename this to `walk_ptrs_ty` or something instead? There's another `walk_ty` in `middle` which is quite useful and could get confusing.